src/api/sync.rs

```diff
@@ -343,6 +386,25 @@ impl Api {
         Ok(filename)
     }
 
+    fn download_from(
+        &self,
+        url: &str,
+        current: &mut usize,
+        file: &mut std::fs::File,
+    ) -> Result<(), ApiError> {
+        let range = format!("bytes={current}-");
+        let response = self
+            .client
+            .get(url)
+            .set(RANGE, &range)
+            .call()
+            .map_err(Box::new)?;
+        let reader = response.into_reader();
+        let mut reader = reader::ResumableReader::new(reader, current);
+        std::io::copy(&mut reader, file)?;
```

Comment:
Given this is only linearly increasing, couldn't we just `Seek` into the file ?
Making `ResumableReader` not necessary (less code  is almost always better) ?

Reply:
@McPatate FYI

Reply:
You don't need to SeekFrom::End I think the response `Content-Length` should already give that information.
And even then you don't need it either since you would be using increasing Seek.

Reply:
Ohh I just realized what you meant. Seek::End would be equal to the partial-length, I see, yes that works too (actually wouldn't the cursor already be there naturally given `io::copy` 's nature?

Reply:
The cursor is already there, but we need to get its byte position to plug it into the `RANGE` header. We could also use `SeekFrom::Current(0)` (or the equivalent `.stream_position()`). `SeekFrom::End(0)` would have the advantage, that we could also use it with `.incomplete` files directly once we implement that, however, that would also be easily achieved by just doing it once in the beginning.

`SeekFrom::Current(0)` might have slightly better performance, I would have to check how it is implemented, so we might prefer it.

Or am I missing another method of retrieving the necessary byte offset?

Reply:
Sorry for the delay. Maybe `file.metadata()` then ? I don't know which is the simplest.
Also is there a way to fuse the `retry` loop with the outer "regular" logic?

Reply:
Please also excuse the late response. file.metadata() would require us to first call file.sync_all() if I am not mistaken, so I do not think that it is desirable. I have used stream_position for now (see next commit).
I'll look into combining the logic this evening.